pandas/core/indexes/base.py

```diff
@@ -3923,6 +3923,16 @@ def values(self):
         """
         return self._data.view(np.ndarray)
 
+    @Appender(IndexOpsMixin.array.__doc__)  # type: ignore
+    @property
```

Comment:
this could be a cached_property I *think*?

Reply:
the Index is immutable but not the underlying, hmm.

Reply:
> the Index is immutable but not the underlying, hmm.

That has always been the case for Index?

Reply:
We fixed the issue with cache_readonly docs not working right? If so, +1 for using it as long as we're micro-optimizing.

Reply:
when putting them in the order of 

```
    @cache_readonly
    @Appender(IndexOpsMixin.array.__doc__)  # type: ignore
    def array(self) -> ExtensionArray:
```

it seems to work